sections/designRules.md

```diff
@@ -483,6 +483,135 @@ https://api.example.org/v1/comments/456</pre>
+      </dd>
+      <dt>Rationale</dt>
+      <dd>
+         <p>Providing problem details in a machine-readable format aids automation and debugging. By using a common error format, APIs do not need to define their own or misuse existing HTTP status codes.</p>
+         <div class="example">
+            The following example shows the head and body of a detailed error response.
+            <pre><code class="http">HTTP/1.1 404 Not Found
+Content-Type: application/problem+json</code><code class="json">{
+  "status": 404,
+  "title": "Resource Not Found",
+  "detail": "No building found with id 12345."
+}
+</code></pre>
+         </div>
+      </dd>
+      <dt>How to test</dt>
+      <dd>
+         Verify all responses with status code <code>4xx</code> or <code>5xx</code> have <code>Content-Type</code> set to <code>application/problem+json</code> or <code>application/problem+xml</code> and contain the fields <code>status</code>, <code>title</code>, and <code>detail</code>. Optional fields are <code>type</code> and <code>instance</code>. Verify no additional fields are present.
+      </dd>
+   </dl>
+</div>
+
+<div class="rule" id="/core/invalid-input" data-type="technical">
+  <p class="rulelab">Use status code 400 for invalid input</p>
+  <dl>
+    <dt>Statement</dt>
+    <dd>
+      <p>API requests containing invalid input MUST result in HTTP status code <code>400 Bad Request</code>. This includes syntax errors, missing or invalid query parameters and schema violations for the request payload.</p>
+    </dd>
+    <dt>Rationale</dt>
+    <dd>
+      <p>The semantics of status code <code>400</code> ("the server cannot or will not process the request due to something that is perceived to be a client error") match validation failures more closely than status code <code>422</code>, which historically originates from WebDAV and introduces no added interoperability benefit.</p>
+    </dd>
+    <dt>How to test</dt>
+    <dd>
+      Verify that operations accepting query parameters and/or a request body contain a response with status code <code>400</code>.
+    </dd>
+  </dl>
+</div>
+
+<div class="rule" id="/core/problem-bad-request" data-type="technical">
+  <p class="rulelab">Add specific errors for bad-request responses</p>
+  <dl>
+    <dt>Statement</dt>
+    <dd>
+      <p>Problem details with status code <code>400</code> (Bad Request) MUST include an additional member <code>errors</code> containing an ordered list of validation error objects, as specified below.</p>
+      <p>Each error object MUST contain: <code>in</code>, <code>location</code>, and <code>detail</code>. The optional <code>code</code> field MAY be used for a stable machine-readable rule identifier.</p>
+      <ul>
+        <li><strong><code>in</code></strong> - where the error occurs: <code>body</code> or <code>query</code>.</li>
+        <li><strong><code>location</code></strong> (optional) - a structured locator for the offending value:
```

Comment:
Hierboven is `location` vereist. Moet `(optional)` hier nog staan? Ik stel voor dat we het required maken en dat we `"location": "body"` zetten voor body errors.